packages/core/src/browser/widgets/widget.ts

```diff
@@ -136,27 +139,41 @@ export function addEventListener<K extends keyof HTMLElementEventMap>(
     );
 }
 
-export function addKeyListener<K extends keyof HTMLElementEventMap>(element: HTMLElement, keybinding: Key, action: () => void, ...additionalEventTypes: K[]): Disposable {
+// tslint:disable-next-line:max-line-length
+export function addKeyListener<K extends keyof HTMLElementEventMap>(element: HTMLElement, keysOrKeyCodes: KeysOrKeyCodes, action: (event: KeyboardEvent) => void, ...additionalEventTypes: K[]): Disposable {
     const toDispose = new DisposableCollection();
-    const keyCode = KeyCode.createKeyCode({ first: keybinding });
+    const isAcceptedKeyCode = (actual: KeyCode) => toKeyCodes(keysOrKeyCodes).some(k => k.equals(actual));
     toDispose.push(addEventListener(element, 'keydown', e => {
         const kc = KeyCode.createKeyCode(e);
-        if (kc.equals(keyCode)) {
-            action();
+        if (isAcceptedKeyCode(kc)) {
+            action(e);
             e.stopPropagation();
             e.preventDefault();
         }
     }));
     for (const type of additionalEventTypes) {
         toDispose.push(addEventListener(element, type, e => {
-            action();
+            // tslint:disable-next-line:no-any
+            const event = (type as any)['keydown'];
+            action(event);
             e.stopPropagation();
             e.preventDefault();
         }));
     }
     return toDispose;
 }
 
+export type KeysOrKeyCodes = Key | KeyCode | (Key | KeyCode)[];
```

Comment:
Should not it be in `keybindings.ts`? Functions should go under corresponding namespaces.